src/olympia/accounts/views.py

```diff
@@ -424,6 +425,14 @@ def perform_destroy(self, instance):
                 u'this account, or transfer them to other users.'))
         return super(AccountViewSet, self).perform_destroy(instance)
 
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        self.perform_destroy(instance)
+        response = Response(status=HTTP_204_NO_CONTENT)
+        if instance == request.user:
+            logout_user(request, response)
```

Comment:
I actually wonder if this should contain an `assert` to actually make sure this worked. On the other hand this then might workaround some of the deletion process if they fail.

Reply:
`logout_user` doesn't return anything anyway, so there is nothing to `assert` on.